lms/djangoapps/experiments/utils.py

```diff
@@ -272,8 +254,8 @@ def get_experiment_user_metadata_context(course, user):
                             has_courses_left_to_purchase = True
                         courses_left_to_purchase_price, courses_left_to_purchase_skus = get_program_price_and_skus(
                             courses_left_to_purchase)
-                        courses_left_to_purchase_url = get_program_purchase_url(courses_left_to_purchase_skus,
-                                                                                program_uuid)
+                        courses_left_to_purchase_url = EcommerceService().get_checkout_page_url(
```

Comment:
i think i'm just missing all the context on this, but were we not doing this previously because of the performance issues that happened when using this method.  
if so, is it safe to use this method now?

Reply:
As I read the code in _get_checkout_page_url_, it doesn't call out to the e-commerce service; all it does is compute the URL for the basket page.

Please let me know if I'm missing anything.

Reply:
ah ok, probably the issue someone ran into was with price rather than the url